packages/vite/src/node/server/middlewares/indexHtml.ts

```diff
@@ -102,12 +102,10 @@ const processNodeUrl = (
     originalUrl !== '/' &&
     htmlPath === '/index.html'
   ) {
-    const replacer = (url: string) =>
-      path.posix.join(
-        devBase,
-        path.posix.relative(originalUrl, devBase),
-        url.slice(1)
-      )
+    // #10990 config.base './' or '' will be changed to `/` in dev mode
+    // this function is only run in dev mode and config.base is always to be an absolute path.
+    // we needn't handle config.base './' or ''
```

Comment:
Probably me being extra nit-picky but I think a single comment liek on line 96 would be enough 😬 

```suggestion
    // prefix with base (dev only, base is never relative)
```

Would also be a nice extra to have line 97 use `path.posix.join` to be consistent too.